spec/API_specification/indexing.rst

```diff
@@ -0,0 +1,195 @@
+
+Using a slice to index a single array axis must select ``m`` elements with index values
+
+::
+
+   i, i+k, i+2k, i+3k, ..., i+(m-1)k
+
+where
+
+::
+
+   m = q + r
+
+and ``q`` and ``r`` (``r != 0``) are the quotient and remainder obtained by dividing ``j-i`` by ``k``
+
+::
+
+   j - i = qk + r
+
+such that
+
+::
+
+   j > i + (m-1)k
+
+.. note::
+    For ``i`` on the interval ``[0, n)`` (where ``n`` is the axis size), ``j`` on the interval ``(0, n]``, ``i`` less than ``j``, and positive step ``k``, a starting index ``i`` is **always** included, while the stopping index ``j`` is **always** excluded. This preserves ``x[:i]+x[i:]`` always being equal to ``x``.
+
+.. note::
+   Using a slice to index into a single array axis should select the same elements as using a slice to index a Python list of the same size.
+
+Slice syntax must have the following defaults. Let ``n`` be the axis (dimension) size.
+
+- If ``k`` is not provided (e.g., ``0:10``), ``k`` must equal ``1``.
+- If ``k`` is greater than ``0`` and ``i`` is not provided (e.g., ``:10:2``), ``i`` must equal ``0``.
+- If ``k`` is greater than ``0`` and ``j`` is not provided (e.g., ``0::2``), ``j`` must equal ``n``.
+- If ``k`` is less than ``0`` and ``i`` is not provided (e.g., ``:10:-2``), ``i`` must equal ``n-1``.
+- If ``k`` is less than ``0`` and ``j`` is not provided (e.g., ``0::-2``), ``j`` must equal ``-n-1``.
+
+Using a slice to index a single array axis must adhere to the following rules. Let ``n`` be the axis (dimension) size.
+
+- If ``i`` equals ``j``, a slice must return an empty array, whose axis (dimension) size along the indexed axis is ``0``.
+
+- Indexing via ``:`` and ``::`` must be equivalent and have defaults derived from the rules above. Both ``:`` and ``::`` indicate to select all elements along a single axis (dimension).
+
+.. note::
+   This specification does not require "clipping" out-of-bounds slice indices. This is in contrast to Python slice semantics where ``0:100`` and ``0:10`` are equivalent on a list of length ``10``.
+
+The following ranges for the start and stop values of a slice must be supported. Let ``n`` be the axis (dimension) size being sliced. For a slice ``i:j:k``, the behavior specified above should be implemented for the following:
+
+- ``i`` or ``j`` omitted (``None``).
+- ``-n <= i <= n``.
+- For ``k > 0`` or ``k`` omitted (``None``), ``-n <= j <= n``.
+- For ``k < 0``, ``-n - 1 <= j <= max(0, n - 1)``.
+
+The behavior outside of these bounds is unspecified.
+
+.. note::
+   *Rationale: this is consistent with bounds checking for integer indexing; the behavior of out-of-bounds indices is left unspecified. Implementations may choose to clip (consistent with Python* ``list`` *slicing semantics), raise an exception, return junk values, or some other behavior depending on device requirements and performance considerations.*
```

Comment:
It was like this before, but I don't understand why this not is in italics. It is already in a note box, I don't think it needs to be italicized.

Reply:
Oh I guess "Rationale" is always italicized wherever it appears in the spec.